extension/src/popup/views/Account/hooks/useGetAccountData.tsx

```diff
@@ -216,24 +222,70 @@ function useGetAccountData(options: {
     const resolvedData = state.data;
 
     const interval = setInterval(async () => {
-      const publicKey = resolvedData.publicKey;
-      const networkDetails = resolvedData.networkDetails;
-      const balancesResult = await fetchBalances(
-        publicKey,
-        _isMainnet,
-        networkDetails,
-        false,
-      );
+      try {
+        const publicKey = resolvedData.publicKey;
+        const networkDetails = resolvedData.networkDetails;
+        const balancesResult = await fetchBalances(
+          publicKey,
+          _isMainnet,
+          networkDetails,
+          false,
+        );
 
-      const payload = {
-        ...state.data,
-        balances: balancesResult,
-      } as AccountData;
-      dispatch({ type: "FETCH_DATA_SUCCESS", payload });
+        const payload = {
+          ...state.data,
+          balances: balancesResult,
+        } as AccountData;
+        dispatch({ type: "FETCH_DATA_SUCCESS", payload });
+      } catch (error) {
+        captureException(`Error refreshing balances on Account - ${error}`);
+      }
     }, 30000);
     return () => clearInterval(interval);
   }, [_isMainnet, state.data, fetchBalances]);
 
+  useEffect(() => {
+    // if it's been 2 minutes since the last balance update, force update
+    if (!state.data || state.data.type === AppDataType.REROUTE) {
+      return;
+    }
+    const resolvedData = state.data;
+
+    const publicKey = resolvedData.publicKey;
+    const networkDetails = resolvedData.networkDetails;
+
+    const refreshBalances = async () => {
+      try {
+        const balancesResult = await fetchBalances(
+          publicKey,
+          _isMainnet,
+          networkDetails,
+          false,
+        );
+
+        const payload = {
+          ...state.data,
+          balances: balancesResult,
+        } as AccountData;
+        dispatch({ type: "FETCH_DATA_SUCCESS", payload });
+      } catch (error) {
+        captureException(
+          `Error refreshing cache balances on Account - ${error}`,
+        );
+      }
+    };
+
+    if (
+      cachedBalances[networkDetails.network]?.[publicKey]?.updatedAt &&
+      cachedBalances[networkDetails.network]?.[publicKey]?.updatedAt <
+        Date.now() - 120000
```

Comment:
if it's been 2 minutes or more since a user has refreshed, we might have some stale data. Let's refresh without having to wait 30 seconds